src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,53 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int GROWTH_FACTOR = 2;
```

Comment:
Consider making `GROWTH_FACTOR` configurable by adding it as a constructor parameter with a default value. This would make the class more flexible, allowing users to control the growth behavior of the stack.